test/perf_test.py

```diff
@@ -0,0 +1,343 @@
+import time
+import unittest
+import select
+import datetime
+
+import paramiko
+
+
+server_list = {
+        "controller": {
+                "ipaddr": "10.10.10.18",
+                "user": "root",
+                "password": "tester"
+            },
+        "node1": {
+                "ipaddr": "10.10.10.14",
+                "user": "root",
+                "password": "tester",
+                "vm_ip": "169.254.0.6"
+            },
+        "node2": {
+                "ipaddr": "10.10.10.95",
+                "user": "root",
+                "password": "tester",
+                "vm_ip": "169.254.0.13"
+            }
+        }
+
+test_VMs = ["VM2","VM3"]
+
+
+class TFVPPVTestCase(unittest.TestCase):
+
+    def _get_VMs_list(self, controller_client):
+        stdin, stdout, stderr = controller_client.exec_command("nova list", timeout=30)
+        
+        stdout_lines = stdout.readlines()
+        stderr_lines = stderr.readlines()
+        
+        if len(stdout_lines) == 0 or len(stderr_lines) > 0:
+
+            for line in stderr_lines: 
+                print line
+
+            raise BaseException("Execute 'nova list' error")
+        
+        vm_list = []
+        
+        for line in stdout_lines[3:-1]: 
+            vm_fields = line.split("|")
+            vm = {"name": vm_fields[2].strip(),
+                  "status": vm_fields[3].strip()}
+            vm_list.append(vm)
+
+        return vm_list
+
+
+    def _start_VM(self, controller_client, vm_name, skip_err=True):
+        stdin, stdout, stderr = controller_client.exec_command("nova start " + vm_name)
+
+        stdout_lines = stdout.readlines()
+        stderr_lines = stderr.readlines()
+        
+        if len(stdout_lines) == 0 or len(stderr_lines) > 0:
+
+            for line in stderr_lines: 
+                print line
+
+            if skip_err == False:
+                raise BaseException("Execute 'nova start' error, vm=" + vm_name)
+
+        print stdout_lines
+
+
+    def _stop_VM(self, controller_client, vm_name, skip_err=True):
+        stdin, stdout, stderr = controller_client.exec_command("nova stop " + vm_name)
+
+        stdout_lines = stdout.readlines()
+        stderr_lines = stderr.readlines()
+        
+        if len(stdout_lines) == 0 or len(stderr_lines) > 0:
+
+            for line in stderr_lines: 
+                print line
+
+            if skip_err == False:
+                raise BaseException("Execute 'nova stop ' error, vm=" + vm_name)
+
+        print stdout_lines
+
+
+    def _update_vrouter(self, node_client, vrouter_path):
+        sftp = node_client.open_sftp()
+        sftp.put(vrouter_path, './contrail-vrouter-dpdk-ci')
+        sftp.chmod('./contrail-vrouter-dpdk-ci', 755)
+        print "Start to update vrouter binary" 
+        stdin, stdout, stderr = node_client.exec_command("./change_vrouter_cores.sh contrail-vrouter-dpdk-ci 10-17", timeout=120)
```

Comment:
Add a #TODO comment to generalize 10-17 in the future for other hardware platform.